plugins/hydra_rq_launcher/hydra_plugins/hydra_rq_launcher/_core.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+import time
+import uuid
+from pathlib import Path
+from typing import Any, Dict, List, Sequence
+
+import cloudpickle
+from fakeredis import FakeStrictRedis
+from hydra.core.config_loader import ConfigLoader
+from hydra.core.hydra_config import HydraConfig
+from hydra.core.singleton import Singleton
+from hydra.core.utils import (
+    JobReturn,
+    configure_log,
+    filter_overrides,
+    run_job,
+    setup_globals,
+)
+from hydra.types import TaskFunction
+from omegaconf import DictConfig, OmegaConf, open_dict
+from omegaconf.errors import ValidationError
+from redis import Redis
+from rq import Queue
+
+from .rq_launcher import RQLauncher
+
+log = logging.getLogger(__name__)
+
+
+def execute_job(
+    idx: int,
+    overrides: Sequence[str],
+    config_loader: ConfigLoader,
+    config: DictConfig,
+    task_function: TaskFunction,
+    singleton_state: Dict[Any, Any],
+    redis_job_id: str,
+) -> JobReturn:
+    """Calls `run_job`
+    """
+    setup_globals()
+    Singleton.set_state(singleton_state)
+
+    sweep_config = config_loader.load_sweep_config(config, list(overrides))
+    with open_dict(sweep_config):
+        sweep_config.hydra.job.id = redis_job_id
+        sweep_config.hydra.job.num = idx
+    HydraConfig.instance().set_config(sweep_config)
+
+    ret = run_job(
+        config=sweep_config,
+        task_function=task_function,
+        job_dir_key="hydra.sweep.dir",
+        job_subdir_key="hydra.sweep.subdir",
+    )
+
+    return ret
+
+
+def launch(
+    launcher: RQLauncher, job_overrides: Sequence[Sequence[str]], initial_job_idx: int,
+) -> JobReturn:
+    """
+    :param job_overrides: a List of List<String>, where each inner list is the arguments for one job run.
+    :param initial_job_idx: Initial job idx in batch.
+    :return: an array of return values from run_job with indexes corresponding to the input list indexes.
+    """
+    setup_globals()
+    assert launcher.config is not None
+    assert launcher.config_loader is not None
+    assert launcher.task_function is not None
+
+    configure_log(launcher.config.hydra.hydra_logging, launcher.config.hydra.verbose)
+    sweep_dir = Path(str(launcher.config.hydra.sweep.dir))
+    sweep_dir.mkdir(parents=True, exist_ok=True)
+
+    # RQ configuration
+    rq_cfg = launcher.rq
+
+    # Redis configuration
+    redis_cfg = launcher.rq["redis"]
```

Comment:
try:
```
 launcher.rq.redis
```

Reply:
I'd prefer this as well, however, I can't figure out how to get this to work. `launcher.rq` is a `dict` rather than `DictConfig` object. Interestingly, `launcher.config.hydra.launcher.params` is a `DictConfig`, but using this rather than `launcher.rq` seems hacky (?). Maybe you have a suggestion?